### PR TITLE
Remove some unused options

### PR DESCRIPTION
**Pull request by fxcoudert**
Following discussion at #31510
Removing options that are never, or very seldom used.

### CODE REVIEW

**Comment by fxcoudert** [comment]
poke @MikeMcQuaid

File: Formula/aubio.rb [hunk 1]
```diff
@@ -14,15 +14,12 @@ class Aubio < Formula
...
   depends_on :macos => :lion
-  depends_on "pkg-config" => :build
   depends_on "libtool" => :build
+  depends_on "pkg-config" => :build
   depends_on "numpy"
   depends_on "python@2"
-  depends_on "libav" => :optional
-  depends_on "libsndfile" => :optional
-  depends_on "libsamplerate" => :optional
   depends_on "fftw" => :optional
```
Comment (javian) [inline Formula/aubio.rb]: While you are at it - shouldn't these be below the `:build` lines ?

Comment (fxcoudert) [reply Formula/aubio.rb]: Build deps first, then normal ones, then optional is the usual order.

**Review by MikeMcQuaid — APPROVED** [review]
Nice work!

**Pull request merged**